### DISCUSSION TITLE
What is the best way to get corpus for traing?

### DISCUSSION THREAD

**Discussion by qingyun1988** [explosion/spaCy#8567] [Help: Other Questions] [answered]
I saw many people have used config command for training on a model.  Using this way needs to prepare training data(corpus) whose file extension is .spacy which includes some corpus data.

I want to know the best way to get training data. Could someone else tell me what your approach is? Thank you.

**Comment by thomashacker** [comment, marked as answer]
It really depends on what you're trying to do which makes it hard to find the "best" way.

**Reply by qingyun1988** [reply]
I am trying to parse the dependency of sentences.

**Reply by DavidGerva** [reply]
@qingyun1988 can you please clarify what do you mean with "dependency of sentences"? And are you working on/in a specific field or domain like biotechnology, finance... In any case I agree with @thomashacker, is a difficult (or too generic) question.

In general you should try to answer some question to yourself:
- do the already-existent models perform quite well? If no, why?
- do you have "material" to work on? I mean: do you have some texts? if no: you have to find something...
- and then: is there an automatic or semi automatic way that can in some (a lot) cases already find what you're looking for?
- if yes, try do use that way to prepare the data, if not you have to manually tag the corpus

you can find on the web some guides on how to train a spacy model: in those articles it should be quite clear what it means to find train data.
